messaging/artemis-jta/src/test/java/io/quarkus/ts/openshift/messaging/artemisjta/AbstractArtemisTest.java

```diff
@@ -0,0 +1,117 @@
+package io.quarkus.ts.openshift.messaging.artemisjta;
+
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static io.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public abstract class AbstractArtemisTest {
+    @Test
+    @Order(1)
+    public void testPrice() {
+        given().queryParam("fail", "false").body("666")
+                .when()
+                .post("/price").then().statusCode(200);
+
+        await().atMost(10, TimeUnit.SECONDS).untilAsserted(() -> assertThat("testPrice:",
+                given()
+                        .when()
+                        .get("/price")
+                        .then()
+                        .statusCode(200)
+                        .extract().body().asString()
+                , equalTo("666:666")));
+    }
+
+    @Test
+    @Order(2)
+    public void testJTAPriceFail() {
+        given().queryParam("fail", "true").body("999")
+                .when()
+                .post("/price").then().statusCode(500);
+        Set<String> s = new HashSet<>(10);
+        // Any Awaitility positive person to refactor this for loop? :-)
```

Comment:
Why? It doesn't wait for anything. It just waits 1 second between iterations, which is probably a useless excersise :-)